### PR TITLE
Document split{_ascii,}_whitespace() for empty strings

### PR DESCRIPTION
**Pull request by vojtechkral**
doc change only

### CODE REVIEW

File: library/core/src/str/mod.rs [hunk 1]
```diff
@@ -946,6 +952,12 @@ impl str {
     ///
     /// assert_eq!(None, iter.next());
     /// ```
+    ///
+    /// If the string is empty or all ASCII whitespace, the iterator yields no string slices:
+    /// ```
+    /// assert_eq!("".split_whitespace().next(), None);
+    /// assert_eq!("   ".split_whitespace().next(), None);
```
Comment (eminence) [inline library/core/src/str/mod.rs:959]: This doc comment is attached to `split_ascii_whitespace` but is mentioning `split_whitespace`

```suggestion
    /// assert_eq!("".split_ascii_whitespace().next(), None);
    /// assert_eq!("   ".split_ascii_whitespace().next(), None);
```

Comment (thomcc) [reply library/core/src/str/mod.rs:959]: Nice catch. I've rejected the PR but will fix it manually if my rejection is too late.

Comment (vojtechkral) [reply library/core/src/str/mod.rs:959]: My bad.

Comment (vojtechkral) [reply library/core/src/str/mod.rs]: Updated...

**Pull request merged**